.automation/arch/build_arch_diagram.py

```diff
@@ -0,0 +1,74 @@
+# !/usr/bin/env python3
+"""
+Generate architecture diagram of MegaLinter
+See how to do it at https://kroki.io/#how
+"""
+import base64
+import tempfile
+import zlib
+from string import Template
+
+import requests
+
+
+class BuildArchDiagram:
+    GATEWAY_SERVER = "https://kroki.io"
+    DIAGRAM_TEMPLATE = "docs/assets/archi.blockdiag.txt"
+
+    def __init__(
+        self,
+        descriptor_language_count,
+        descriptor_format_count,
+        descriptor_tooling_format_count,
+    ):
+        self.diagram_dict = dict(
+            language=str(descriptor_language_count),
+            format=str(descriptor_format_count),
+            tooling_format=str(descriptor_tooling_format_count),
+        )
+
+        self.build_diagram()
+
+    # Generate binary image following specified format
+    def get_image_bin(self, format="svg", tempfile="kroki"):
```

Comment:
`get_image_file` might be a clearer name for this method, because `get_image_bin` sounds like it parallels `get_image_text` and returns something like `image_content`, rather than `output_file`.